src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,128 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Deals with all Storage operations.
+ */
+public class Storage {
+    private static File memoryFile;
+    private Path path;
+
+
+    public Storage(String filePath) {
+        path = Paths.get(filePath);
+    }
+
+    private List<Task> parseData(File f) throws FileNotFoundException {
+        List<Task> tasks = new ArrayList<>();
+
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String str = s.nextLine();
+            String[] args = str.split(" \\| ");
+            boolean done = args[1].equals("1");
+            String description = args[2];
```

Comment:
Might be better to give more descriptive names to `str` and `args` as they are used throughout this method for better readability. Also consider changing `done` to `isDone` to follow the coding standard.